File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x<0 || (x%10==0 && x!=0)) return false;
        int copy=x,digit;
        long long rev=0;
        while(x!=0){
            digit=x%10;
            rev=(rev*10)+digit;
            x=x/10;
        }
        if(copy==rev) return true;
        else return false;
    }
};

File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x<0 or x%10==0 and x!=0: return False
        rev=0; copy=x
        while x!=0:
            ld=x%10
            rev=rev*10+ld
            x=x//10
        return copy==rev
